@app/server/src/plugins/CreateUploadUrlPlugin.ts

```diff
@@ -0,0 +1,37 @@
+import { makeExtendSchemaPlugin, gql } from "graphile-utils";
+import uuidv4 from "uuid/v4";
+import * as aws from "aws-sdk";
+import { awsRegion, uploadBucket } from "@app/config";
+
+const CreateUploadUrlPlugin = makeExtendSchemaPlugin(() => ({
+  typeDefs: gql`
+    extend type Mutation {
+      """
+      Get a signed URL for uploading files. It will expire in 60 seconds.
+      """
+      createUploadUrl(contentType: String): String!
```

Comment:
Please use an input object and payload as described in [Caleb's brilliant article](https://blog.apollographql.com/designing-graphql-mutations-e09de826ed97). The payload is important for making it so you can enhance the mutation in future with additional fields; the input object in this case would be just for consistency, but it's useful to pass bundles of arguments around in a single object as the number of arguments get larger.

Reply:
The exclamation points are probably the wrong way round here - content type should be required (?), but we do not guarantee a non-null result (e.g. if an error is thrown). See [why is it nullable - root fields](https://www.graphile.org/postgraphile/why-nullable/#root-query-mutation-and-subscription-fields).

Reply:
Thinking about content type... Should this be an enum? We probably only want people to upload certain classes of files, rather than arbitrary files? Also the enum could be associated internally with different limits - e.g. an avatar would have small size, whereas a report or CV might have a larger file size limit. Also, they'd probably get stored to slightly different places.